### ISSUE TITLE
[Data Table]: accessibility issues related to the component

### ISSUE THREAD

**Issue by NatalliaAlieva** [epam/UUI#2992]
### Description
Open columns configuration modal button has no accessible name.

### Steps to Reproduce

1. Open Data Table component with 'showColumnsConfig={true}', 'allowColumnsReordering={true}' props or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Use the Tab key to navigate to table.
3. Use the Tab key to navigate to [Open columns configuration modal] button and note the announcement.

### Actual result
![2024-06-05_16h08_27](https://github.com/epam/UUI/assets/70166849/c0428343-3739-450d-a1e3-b96209e268dd)

### Expected result
Open columns configuration modal button should be announced with accessible name that clearly describes its purpose.

Failed WCAG Checkpoint: 4.1.2 Name, Role, Value (Level A)
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

### Description
When User selects sorting direction in dropdown menu for any column, then there is no information about it for screen reader.

### Steps to Reproduce

1. Open Data Table component with filtersConfig passed as prop or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Open filter dropdown.
3. Select either "Sort ascending" or "Sort descending" item and note the announcement.

### Actual result
![table-sort](https://github.com/epam/UUI/assets/70166849/5a918a29-6cd3-4aff-9b36-41f6e8922d89)
There is no difference for screen reader either sorting item is selected or not.

### Expected result
It should be clear for screen reader if any sorting direction is selected.

### Description
The Data Table does not have accessible name.

### Steps to Reproduce

1. Open Data Table component or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Use the Tab key to navigate to table and note the announcement.
3. Use the Tab key to navigate to [Open columns configuration modal] button and note the announcement.

### Actual result
![2024-06-24_12h29_01](https://github.com/epam/UUI/assets/70166849/b666d225-3eca-4b14-8de2-3a673ee601b8)
![table-name](https://github.com/epam/UUI/assets/70166849/868dc584-3a07-4208-a3ab-d1a090263b30)

The table does not have accessible name. Aria-label is missed.

### Expected result
The table has accessible name.

Example of the correct behavior:
https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/table/

[Success Criterion 4.1.2 Name, Role, Value (level A).](https://www.w3.org/WAI/WCAG22/quickref/?showtechniques=211#name-role-value)

### Description
Tooltip text "Column header name" (on columns' names) disappears when the pointer moved over it.

### Steps to Reproduce

1. Open Data Table component or Demo (Filtered table) (https://uui.epam.com/demo?id=table).
2. Hover mouse pointer over column's header -> tooltip text (column header name) is displayed.
3. Hover mouse pointer over tooltip text and observe.

### Actual result

Tooltip text with column header name  disappears when the pointer moved over it.

### Expected result
The pointer can be moved over the tooltip text without the tooltip text disappearing.
Failed WCAG Checkpoint: 1.4.13 Content on Hover or Focus (Level AA)
https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html

## Environment
* UUI version:  6.1.4
* Browser name: Crome latest
* Operating System: Windows latest

**Comment by JuliaMV** [comment]
Released in v6.4.4.
